tests/crypto/rand32/src/main.c

```diff
@@ -30,15 +30,17 @@
 
 void test_rand32(void)
 {
-	uint32_t gen, last_gen;
+	uint32_t gen, last_gen, tmp;
 	int rnd_cnt;
 	int equal_count = 0;
 	uint32_t buf[N_VALUES];
 
 	/* Test early boot random number generation function */
+	/* Add a condition about length less than uint32 */
+	z_early_boot_rand_get((uint8_t *)&tmp, (size_t)1);
 	z_early_boot_rand_get((uint8_t *)&last_gen, sizeof(last_gen));
 	z_early_boot_rand_get((uint8_t *)&gen, sizeof(gen));
-	zassert_true(last_gen != gen,
+	zassert_true(last_gen != gen && last_gen != tmp && tmp != gen,
```

Comment:
Hi, does  "zassert_true(last_gen != gen && last_gen != tmp)" also get the same result we want? I guess check these two condition is enough.

Reply:
It cannot make sure if `gen` equal `tmp` by your comment.

Reply:
Got it! Thanks!